### ISSUE TITLE
feature conflicts?

### ISSUE THREAD

**Issue by sphaero** [jvcleave/ofxImGui#87]
I'm running into issues when using information from ImGui sources and using them with this addon. For example copying themes from: https://github.com/ocornut/imgui/issues/707 seems to conflict with the BaseTheme design in this addon. It overrides some assumed defaults coming from ImGui and therefore the themes don't render the same. Implementing an inherited BaseTheme class doesn't work because the colors are minimalised into 5 colors and the updateColors isn't virtual. 

I could change this all of course but it seems the addon is adding features while blocking other features? As ImGui is in constant flux I would suggest to keep the addon close as possible to ImGui. A theme would be no more than just a method which sets some ImGuiStyle properties? Or use a serializer (i.e. JSON)? Or I might be missing something?

**Comment by jvcleave** [comment]
Yeah the idea for the BaseTheme was to be able to extend it and drop in your own themes. However it's pretty old code and I personally haven't used it much (I just use the defaults). 

Making `updateColors` virtual seems pretty minimal. Would that fix the issue?

I would think something that extends BaseTheme could have however many colors it wanted.

**Comment by jvcleave** [comment]
I pushed https://github.com/jvcleave/ofxImGui/commit/3776f4c81f31e7473225e4a2999823ef6dcbebd2 that changes things a bit

- `BaseTheme` is simplified - it's basically an interface that just requires you to call `setup` so you can use whatever colors you want

- ofxImGui uses `DefaultTheme` that had the styling that was previously in `BaseTheme`

- Added a Custom theme demo in `example-demo` (see the `example-demo/MyTheme` class)

- removed `BaseTheme:: themeColorsWindow`. This would probably fit better in an example
